spec/controllers/orders_controller_spec.rb

```diff
@@ -0,0 +1,51 @@
+require "rails_helper"
+require "pry"
+RSpec.describe OrdersController, type: :controller do
+  let!(:category){FactoryBot.create :category}
```

Comment:
phân biệt let và let!

Reply:
let nó sẽ khai báo lần đầu tiên và mỗi it bên dưới thì nó sẽ lấy luôn giá trị khởi tạo ban đầu, let! nó sẽ tính toán lại và lấy giá trị mới. VD: khi tạo 1 biến có giá trị Time.now thì mình sẽ dùng let! sẽ trả về đúng thời điểm it chạy còn let nó chỉ lấy thời gian khởi tạo khi chạy file test ạ

Reply:
là sao em cái let em sang test case khác thì còn là gía trị đó không em

Reply:
@anlvq-1058 tức là không còn là giá trị ban đầu đó anh, mà sẽ execute lại từ đầu

Reply:
let! nó sẽ chạy ngầm bằng 1 khối before trước mỗi test case

Reply:
Bây h em bỏ hết ! ở đây đi xem có bị lỗi không chỗ nào bỏ đc ! thì bỏ đi